libcaf_openssl compile errors on Windows

In 0.15.4, if `CAF_NO_OPENSSL=OFF` (OpenSSL is enabled) Windows 2015 and 2017 fail with errors like:

```
C:\Program Files (x86)\Windows Kits\8.1\Include\shared\ws2def.h(221): error C2011: 'sockaddr': 'struct' type redefinition (compiling source file C:\Users\appveyor\.conan\data\caf\0.15.4\actor-framework\stable\build\f121568f99cbb51abc2e1e9ab27da0d602a74a7d\actor-framework-0.15.4\libcaf_openssl\src\manager.cpp) 

C:\Program Files (x86)\Windows Kits\8.1\Include\um\winsock.h(1002): note: see declaration of 'sockaddr' (compiling source file C:\Users\appveyor\.conan\data\caf\0.15.4\actor-framework\stable\build\f121568f99cbb51abc2e1e9ab27da0d602a74a7d\actor-framework-0.15.4\libcaf_openssl\src\manager.cpp)

C:\Program Files (x86)\Windows Kits\8.1\Include\shared\ws2def.h(421): error C2059: syntax error: 'constant' (compiling source file C:\Users\appveyor\.conan\data\caf\0.15.4\actor-framework\stable\build\f121568f99cbb51abc2e1e9ab27da0d602a74a7d\actor-framework-0.15.4\libcaf_openssl\src\manager.cpp)
```

See [this Appveyor build](https://ci.appveyor.com/project/sourcedelica/conan-caf/build/1.0.30/job/48hnvv3uk81e0tpm) for a full list.

--- a/libcaf_io/caf/io/network/default_multiplexer.hpp
+++ b/libcaf_io/caf/io/network/default_multiplexer.hpp
@@ -59,8 +59,8 @@
 #   define WINVER WindowsVista
 #   include <w32api.h>
 # endif
-# include <windows.h>
 # include <winsock2.h>
+# include <windows.h>
 # include <ws2tcpip.h>
 # include <ws2ipdef.h>
 #else

--- a/libcaf_openssl/test/authentication.cpp
+++ b/libcaf_openssl/test/authentication.cpp
@@ -22,7 +22,14 @@
 #define CAF_SUITE openssl_authentication
 #include "caf/test/unit_test.hpp"
 
-#include <unistd.h>
+#ifndef CAF_WINDOWS
+# include <unistd.h>
+#else
+# include <io.h>
+# include <windows.h>
+# define F_OK 0
+# define PATH_MAX MAX_PATH
+#endif
 
 #include <vector>
 #include <sstream>
@@ -60,10 +67,14 @@ public:
      // TODO: https://github.com/actor-framework/actor-framework/issues/555
      path += "/../../libcaf_openssl/test";
      char rpath[PATH_MAX];
+#ifndef CAF_WINDOWS
      auto rp = realpath(path.c_str(), rpath);
+#else
+     auto rp = GetFullPathName(path.c_str(), PATH_MAX, rpath, nullptr);
+#endif
      std::string result;
      if (rp)
-       result = rp;
+       result = rpath;
      return result;
   }
 };

--- a/libcaf_openssl/test/remote_actor.cpp
+++ b/libcaf_openssl/test/remote_actor.cpp
@@ -56,7 +56,9 @@ struct fixture {
   config client_side_config;
   actor_system client_side{client_side_config};
   fixture() {
+#ifdef CAF_LINUX
     signal(SIGPIPE, SIG_IGN);
+#endif
   }
 };
 

